docs/ref/models/querysets.txt

```diff
@@ -1988,7 +1988,7 @@ Combines two ``QuerySet``\s using the SQL ``AND`` operator.
 The following are equivalent::
 
     Model.objects.filter(x=1) & Model.objects.filter(y=2)
-    Model.objects.filter(x=1, y=2)
+    Model.objects.filter(x=1).filter(y=2)
```

Comment:
As far as I'm aware, the below example with `Q()` objects is also incorrect when filtering against multi-valued relationships :thinking: Maybe we should leave examples as they are and add an admonition about multi-valued relationships with link to the ["Spanning multi-valued relationships"](https://docs.djangoproject.com/en/4.1/topics/db/queries/#spanning-multi-valued-relationships) section :thinking:

Reply:
@felixxm Im not sure i understood your point, but the example `Model.objects.filter(x=1).filter(y=2)` seems to me more universal than `Model.objects.filter(x=1, y=2)`. Why ? Because
`Model.objects.filter(x=1) & Model.objects.filter(y=2) <=> Model.objects.filter(x=1, y=2) <=> Model.objects.filter(x=1).filter(y=2)` are equivalent, but when joins are involved,
its like this:
```
Model.objects.filter(related__x=1).filter(related__y=1) <=> Model.objects.filter(related__x=1) & Model.objects..filter(related__y=1)
# but not
Model.objects.filter(related__x=1, related__y=1) 
# which gives another output
```
so 
```
# with whatever a simple attribute or a join this equivalence is always true:
Model.objects.filter(whatever1=1) & Model.objects.filter(whatever2=2) <=> Model.objects.filter(whatever1=1).filter(whatever2=2) 
# but this equivalent is not always true, if whatever is a join
Model.objects.filter(whatever1=1) & Model.objects.filter(whatever2=2) <=> Model.objects.filter(whatever1=1, whatever2=2) 
```
So it seems to me that the equivalence that is always true should be in the example. But i think we will also need to explain why

Reply:
> Im not sure i understood your point, ...

My point is that `Model.objects.filter(Q(x=1) & Q(y=2))` is also not equivalent to the `Model.objects.filter(x=1) & Model.objects.filter(y=2)` when filtering against multi-valued relationships.

Reply:
@Guilouf query combination logic was implemented a while ago and since JOIN reuse for multi-valued relationship requires the exact same `filter()` call to take place it would have required extra work to get working in `QuerySet.combine`.

It seems like [this was done intentionnaly](https://github.com/django/django/blob/fcc7dc5781667932bf0bf8bec76df458836e5e95/django/db/models/sql/query.py#L611-L612) though and at this point the behaviour has ossified into code bases and it would be backward incompatible to change.

Reply:
@felixxm An admonition sounds ok however it would probably need to be worded to the effect of "combining querysets with & effectively chains them together"…  Wouldn't that just be more clear by doing the `filter(x).filter(y)` example?  If the Q&Q isn't equivalent perhaps that example should be removed? 🤔

Now that I think about it perhaps the wording should be made more accurate (something along these lines):

```
AND (&)

Combines two QuerySets together by chaining them, having the effect of ANDing.

The following are equivalent:
    Model.objects.filter(x=1) & Model.objects.filter(y=2)
    Model.objects.filter(x=1).filter(y=2)

    (drop the Q example)
```

Then possibly clarifying a point about multi-valued-spanning relationships 🤔

Reply:
> Wouldn't that just be more clear by doing the `filter(x).filter(y)` example?

Agreed.

> If the Q&Q isn't equivalent perhaps that example should be removed? thinking

Yes, showing `Model.objects.filter(Q(x=1)).filter(Q(y=2))` doesn't have much value.
 
> Now that I think about it perhaps the wording should be made more accurate (something along these lines):
> 
> ```
> AND (&)
> 
> Combines two QuerySets together by chaining them, having the effect of ANDing.
> 
> The following are equivalent:
>     Model.objects.filter(x=1) & Model.objects.filter(y=2)
>     Model.objects.filter(x=1).filter(y=2)
> 
>     (drop the Q example)
> ```
> 
> Then possibly clarifying a point about multi-valued-spanning relationships thinking

Sounds good :+1:

Reply:
@felixxm Decided in the end to just say "similar to chaining" … I reread the `filter()` & `__and__()` methods and noted their differences.